src/irr/asset/CMeshManipulator.cpp

```diff
@@ -718,6 +334,19 @@ asset::ICPUMeshBuffer* CMeshManipulator::createMeshBufferUniquePrimitives(asset:
 	return clone;
 }
 
+//
+asset::ICPUMeshBuffer* CMeshManipulator::calculateSmoothNormals(asset::ICPUMeshBuffer* inbuffer, float epsilon,
+	asset::E_VERTEX_ATTRIBUTE_ID normalAttrID, VxCmpFunction vxcmp) const
+{
+	if (inbuffer == nullptr)
```

Comment:
you also need to check whether the index type is `EIT_UNKNOWN` 
(that the mesh has no shared vertices)